skfp/model_selection/scaffold_split.py

```diff
@@ -0,0 +1,365 @@
+    to the time split, assuming that new molecules (test set) will be structurally different in terms of
+    scaffolds from the training set.
+
+    This approach is known to have certain limitations. In particular, disconnected molecules or
+    molecules with no rings will not get a scaffold, resulting in them being grouped together
+    regardless of their structure.
+
+    The split is fully deterministic, with the smallest scaffold sets assigned to the test
+    subset and the rest to the training subset.
+
+    The split fractions (train_size, test_size) must sum to 1.
+
+    Parameters
+    ----------
+    data : sequence
+        Sequence representing either SMILES strings or RDKit `Mol` objects.
+
+    additional_data: list[sequence]
+        Additional sequences to be split alongside the main data (e.g., labels or feature vectors).
+
+    train_size : float, default=None
+        The fraction of data to be used for the train subset. If None, it is set to 1 - test_size.
+        If test_size is also None, it will be set to 0.8.
+
+    test_size : float, default=None
+        The fraction of data to be used for the test subset. If None, it is set to 1 - train_size.
+        If train_size is also None, it will be set to 0.2.
+
+    include_chirality: bool, default=False
+        Whether to take chirality of molecules into consideration.
+
+    return_indices : bool, default=False
+        Whether the method should return the input object subsets, i.e. SMILES strings
+        or RDKit `Mol` objects, or only the indices of the subsets instead of the data.
+
+    Returns
+    ----------
+    subsets : tuple[list, list, ...]
+    Tuple with train-test subsets of provided arrays. First two are lists of SMILES strings or RDKit `Mol` objects,
+    depending on the input type. If `return_indices` is True, only lists of indices are returned,
+    and any additional data is ignored.
+
+    References
+    ----------
+    .. [1] `Bemis, G. W., & Murcko, M. A.
+        "The properties of known drugs. 1. Molecular frameworks."
+        Journal of Medicinal Chemistry, 39(15), 2887-2893.
+        https://www.researchgate.net/publication/14493474_The_Properties_of_Known_Drugs_1_Molecular_Frameworks`_
+
+    .. [2] `Z. Wu, B. Ramsundar, E. N. Feinberg, J. Gomes, C. Geniesse, A. S. Pappu, K. Leswing, V. Pande
+        "MoleculeNet: A Benchmark for Molecular Machine Learning."
+        Chemical Science, 9(2), 513-530.
+        https://www.researchgate.net/publication/314182452_MoleculeNet_A_Benchmark_for_Molecular_Machine_Learning`_
+
+    """
+    if train_size is None and test_size is None:
+        raise ValueError("Either train_size or test_size must be provided")
+    if (
+        train_size is not None
+        and test_size is not None
+        and not np.isclose(train_size + test_size, 1.0)
+    ):
+        raise ValueError("train_size and test_size must sum to 1.0")
+
+    train_size, test_size = fill_missing_sizes(train_size, test_size)
```

Comment:
1. First `if` clashes with default values and should be removed
2. Second `if` can be checked inside `fill_missing_sizes`